common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -1288,6 +1296,33 @@ def setUp(self):
         for course in self.courses:
             self.assert_enrollment_status(course_id=str(course.id), username=self.USERNAME, is_active=True)
 
+    def _create_test_retirement(self, user=None):
+        """
+        Helper method to create a RetirementStatus with useful defaults
+        """
```

Comment:
Since this only creates the retirement row, but doesn't hash the username in auth_user, it isn't actually testing the case where this runs the second time after already reaching COMPLETED.  Maybe add a boolean arg here to indicate whether or not to hash the username, then use a ddt on tests that use this function to test with and without hashing username?

Reply:
@bmedx: @pwnage101 brings up a good point. When you've been writing these tests elsewhere, has the username-change happened before the second call to check for idempotency?

Reply:
It's fine to test it both ways, but part of the point of using `get_retirement_for_retirement_action` is that it will get the user object by the original username. AFAIK the calls use either the passed in username, original_username, or the user object returned from that call so I don't see it as much of an issue.

Reply:
OK - thanks. Ready to give this PR a final review then?